theano/tensor/basic.py

```diff
@@ -968,6 +968,53 @@ def _pack(x):
         return [x]
 
 
+def check_and_normalize_axes(x, axis):
+    """
+    Check axes, normalize and convert them to a Python list of integers.
+    Return an empty list if argument is None.
+
+    Parameters
+    ----------
+    x: Tensor variable
+    axis = Integer, tuple or list of integers
+
+    Returns
+    -------
+    axis: list of integers
+    """
+    x = as_tensor_variable(x)
```

Comment:
If x is already a Tensor variable (as you have mentioned in the Parameters L#978), why do we need this conversion? Also you do the conversions at L#1600 and L#1431.